cpp/tests/groupby/sort/group_nth_element_test.cu

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <tests/groupby/common/groupby_test_util.hpp>
+
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+#include <tests/utilities/type_lists.hpp>
+
+#include <cudf/detail/aggregation/aggregation.hpp>
+#include <algorithm>
+
+namespace cudf {
+namespace test {
+
+
+template <typename V>
+struct groupby_nth_element_test : public cudf::test::BaseFixture {};
+
+TYPED_TEST_CASE(groupby_nth_element_test, cudf::test::AllTypes);
+
+TYPED_TEST(groupby_nth_element_test, basic)
+{
+    using K = int32_t;
+    using V = TypeParam;
+    using R = experimental::detail::target_type_t<V, experimental::aggregation::NTH_ELEMENT>;
+
+    fixed_width_column_wrapper<K> keys        { 1, 2, 3, 1, 2, 2, 1, 3, 3, 2};
+    fixed_width_column_wrapper<V> vals        { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    //keys                                    { 1, 1, 1, 2, 2, 2, 2, 3, 3, 3};
+    //vals                                    { 0, 3, 6, 1, 4, 5, 9, 2, 7, 8};
+
+    fixed_width_column_wrapper<K> expect_keys { 1, 2, 3 };
+
+    auto agg = cudf::experimental::make_nth_element_aggregation(0);
+    fixed_width_column_wrapper<R> expect_vals0 {0, 1, 2 };
+    test_single_agg(keys, vals, expect_keys, expect_vals0, std::move(agg));
+
+    agg = cudf::experimental::make_nth_element_aggregation(1);
+    fixed_width_column_wrapper<R> expect_vals1 {3, 4, 7 };
+    test_single_agg(keys, vals, expect_keys, expect_vals1, std::move(agg));
+
+    agg = cudf::experimental::make_nth_element_aggregation(2);
+    fixed_width_column_wrapper<R> expect_vals2 {6, 5, 8 };
+    test_single_agg(keys, vals, expect_keys, expect_vals2, std::move(agg));
+}
+
+TYPED_TEST(groupby_nth_element_test, basic_out_of_bounds)
```

Comment:
Add a test for negative indices too. This will ensure that you also have implemented `groupby.last()`.